.github/workflows/merge-branch.yml

```diff
@@ -0,0 +1,49 @@
+name: Merge - Merge Source Branch to Target Branch
+on:
+  workflow_call:
+    inputs:
+      componentName:
+        description: The component name using the workflow
+        required: true
+        type: string
+      sourceBranch:
+        description: Where the merge code is coming from
+        required: true
+        type: string
+      targetBranch:
+        description: Where the code is going
+        required: true
+        type: string
+jobs:
+  merge:
+    runs-on: ubuntu-latest
+    name: Merge
+    steps:
+      # Checkout Calling Repo Target Branch
+      - name: Checkout Calling Repo
+        uses: actions/checkout@v4
+        with:
+          repository: SecureApiGateway/${{ inputs.componentName }}
+          ref: ${{ inputs.targetBranch }}
+      - name: Set Git config
+        run: |
+          git config user.name fropenbanking
+          git config user.email obst@forgerock.com
+      - name: Perform Merge
+        run: |
+          git merge --no-ff ${{ inputs.sourceBranch }} -m "Auto-merge ${{ inputs.sourceBranch }} into ${{ inputs.targetBranch }} Branch" || echo "Conflicts found, unable to merge"
```

Comment:
I think it is better to write this as an if statement, then if the merge fails we can exit early with an error code.

Otherwise, we will continue and hit the push which will fail with a very confusing error msg

Reply:
Also lets use git rebase to begin with, see how that goes. Otherwise we end up with the merge commits appearing in the ob-v4 branch each time we merge in.